### ISSUE TITLE
请问这是什么问题呢

### ISSUE THREAD

**Issue by MOONWZ** [ducnt9907/ColonFormer#1]
![image](https://user-images.githubusercontent.com/74870759/204106320-6daf00f1-a8fa-4028-a083-18412ff236e4.png)
预训练模型是mit_b3.pth，模型文件使用的提供的ColonFormerS_exp1_run3.pth
请问有设么配置不同么，为什么我从您提供的模型文件运行test.py会出现这样的问题

![image](https://user-images.githubusercontent.com/74870759/204106358-91222845-551c-4a2c-93a6-dc5b8fd2dd71.png)
在运行train.py可以正常运行但是也出现了类似的bug，关于head.weight和head.bias，请问是还有什么需要配置的地方么

**Comment by MOONWZ** [comment]
model.load_state_dict(checkpoint['state_dict'], strict=False)
model = nn.DataParallel(model)
我加了这两行之后可以运行，但是很明显有问题，我是在kvasir数据集上测试
scores: dice=0.021817205025081388, miou=0.011018909668128631, precision=0.9999999998771499, recall=0.011028913638442635

**Comment by ducnt9907** [comment]
The ColonFormerS in ColonFormerS_exp1_run3.pth use backbone MIT-B2 so you should use mit_b2.pth instead of mit_b3.pth

**Comment by MOONWZ** [comment]
![image](https://user-images.githubusercontent.com/74870759/204119641-a230f38c-7b7b-4b20-a867-022e8dfc1e32.png)
![image](https://user-images.githubusercontent.com/74870759/204119658-4a577732-2ab6-4919-a517-5616d9b381f0.png)
![image](https://user-images.githubusercontent.com/74870759/204119787-7bb7965b-592a-4339-bb15-06c3db3ed918.png)
![image](https://user-images.githubusercontent.com/74870759/204119767-ef053d15-9d9f-4aea-89bd-c8c69bc0a5c7.png)
I also have a problem with the setting of experimental data set. It is written in the paper that experiment 1 uses 90% of Kvasir and ClinicDB as the training set, and tests with the test sets of other five data sets. However, it is impossible to get the same experimental results in this way, because the test set is carried out on the known data set. I could not get such effect in the picture CVC-ColonDB CVC-T tis-Larib using the model file you provided. May I ask how the training test set was set for the experiment in Table 1 of your paper? The premise is that I used the data in the download link of the data set provided by you, so the Settings are the same. Your paper is very meaningful to me. Thank you for your help.

**Comment by MOONWZ** [comment]
> ColonFormerS_exp1_run3.pth 中的 ColonFormerS 使用主干 MIT-B2，因此您应该使用 mit_b2.pth 而不是 mit_b3.pth

![image](https://user-images.githubusercontent.com/74870759/204119641-a230f38c-7b7b-4b20-a867-022e8dfc1e32.png)
![image](https://user-images.githubusercontent.com/74870759/204119658-4a577732-2ab6-4919-a517-5616d9b381f0.png)
![image](https://user-images.githubusercontent.com/74870759/204119787-7bb7965b-592a-4339-bb15-06c3db3ed918.png)
![image](https://user-images.githubusercontent.com/74870759/204119767-ef053d15-9d9f-4aea-89bd-c8c69bc0a5c7.png)
I also have a problem with the setting of experimental data set. It is written in the paper that experiment 1 uses 90% of Kvasir and ClinicDB as the training set, and tests with the test sets of other five data sets. However, it is impossible to get the same experimental results in this way, because the test set is carried out on the known data set. I could not get such effect in the picture CVC-ColonDB CVC-T tis-Larib using the model file you provided. May I ask how the training test set was set for the experiment in Table 1 of your paper? The premise is that I used the data in the download link of the data set provided by you, so the Settings are the same. Your paper is very meaningful to me. Thank you for your help.

**Comment by ducnt9907** [comment]
Sorry about my mistake, the pretrained weight should be in experiment 3.1 instead of experiment 1, I will update the link of the dataset so you could reproduce results in the paper. Many thanks.

**Comment by MOONWZ** [comment]
> Sorry about my mistake, the pretrained weight should be in experiment 3.1 instead of experiment 1, I will update the link of the dataset so you could reproduce results in the paper. Many thanks.

Thank you very much. This will bring great convenience to my future experimental work.

**Comment by ducnt9907** [comment]
> > Sorry about my mistake, the pretrained weight should be in experiment 3.1 instead of experiment 1, I will update the link of the dataset so you could reproduce results in the paper. Many thanks.
> 
> Thank you very much. This will bring great convenience to my future experimental work.

Now you can try again with the setting in Experiment 3.1
